### PR TITLE
Ginkgo upstream enhancements

### PR DESCRIPTION
**Pull request by pohly**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

- restore JUnit writing inside -report-dir
- restore e2e/framework/log_test.go

#### Special notes for your reviewer:

@chendave: you should be able to fixup + squash during a rebase. The revert commit and the reverted commit need to be deleted.

### CODE REVIEW

**Comment by chendave** [comment]
@pohly  granted you the write permission on this fork repo, so that you can push your enhancement directly to: https://github.com/kubernetes/kubernetes/pull/109111 if you want in the future. I will update that pr in this week later, thanks for your help on this!

**Comment by pohly** [comment]
Note that I dropped the commit with the `WHAT=ginkgo`. I'll directly push commits in the future, but will let you handle the rebasing because that'll give you a chance to review and is less risky - a single person should be in charge of overwriting your branch.

File: test/e2e/framework/log_test.go [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+		framework.Logf("after")
+		framework.ExpectEqual(true, false, "true is never false either")
+	})
+})
+
+func TestFailureOutput(t *testing.T) {
+	// Run the Ginkgo suite with output collected by a custom
+	// reporter in adddition to the default one. To see what the full
+	// Ginkgo report looks like, run this test with "go test -v".
+	config.DefaultReporterConfig.FullTrace = true
```
Comment (chendave) [inline test/e2e/framework/log_test.go]: `config.DefaultReporterConfig.FullTrace` is not available for v2.

File: test/e2e/framework/log_test.go [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+	gomega.RegisterFailHandler(framework.Fail)
+	fakeT := &testing.T{}
+	reporter := reporters.NewFakeReporter()
```
Comment (chendave) [inline test/e2e/framework/log_test.go]: there is no such fake reporter for v2, this is a custom reporter for v1.

File: test/e2e/framework/log_test.go [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+	commonOutput := `
+
+INFO: after
+FAIL: true is never false either
+Expected
+    <bool>: true
+to equal
+    <bool>: false
+
+Full Stack Trace
```
Comment (chendave) [inline test/e2e/framework/log_test.go:110]: The stack trace here might need to be updated as well, I haven't got a chance to verify it.

File: test/e2e/framework/log_test.go [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+	log_test.go:71
+k8s.io/kubernetes/test/e2e/framework_test.runTests()
+	log_test.go:47
+
+`
+
+	// Sorted by name!
+	expected := suiteResults{
+		testResult{
+			name: "[Top Level] log asserts",
```
Comment (chendave) [inline test/e2e/framework/log_test.go]: IIRC, there is not such `[Top Level]` in v2.

**Review by chendave — COMMENTED** [review]
more work to do or investigate if we want to revert this. I'd prefer to deferring this.

**Review by chendave — COMMENTED** [review]
sorry, I missed that you have already fixed those.

**Pull request closed without merging**